app/src/androidTest/kotlin/org/wordpress/aztec/demo/tests/SimpleTextFormattingTests.kt

```diff
@@ -310,4 +306,37 @@ class SimpleTextFormattingTests : BaseTest() {
                 .verifyHTML(expected2)
     }
 
+
+    // Test reproducing the issue described in
+    // https://github.com/wordpress-mobile/AztecEditor-Android/pull/466#issuecomment-322404363
+    @Test
+    fun testInlineStyleAndDelete() {
+        val text1 = "some"
+        val html = "<i>som</i>"
+
+        EditorPage()
+                .toggleItalics()
+                .insertText(text1)
+                .delete(1)
+                .toggleHtml()
+                .verifyHTML(html)
+    }
+
+    // Test reproducing the issue described in
+    // https://github.com/wordpress-mobile/AztecEditor-Android/pull/466#issuecomment-322405856
+    @Test
+    fun testInlineStyleAndSpace() {
+        val text1 = "some"
+        val text2 = "text"
+        val text3 = " "
```

Comment:
Any reason not to combine `text2` and `text3` into a single string `text ` (with the space at the end of that string, rather than entering the space separately)?

Reply:
Keyboard is dismissed between two calls to `insertText`, honestly I'm not sure if that changes anything. I tried both methods and they both pass.

Reply:
Ok. Not dismissing the keyboard seems closer to the behavior of a user entering text in the editor, but given that it doesn't make a difference to the test I don't have a strong opinion about it. :)

Reply:
Fixed in ee5dde0